xdsl/transforms/convert_onnx_to_linalg.py

```diff
@@ -34,6 +36,37 @@ def match_and_rewrite(self, add: onnx.Add, rewriter: PatternRewriter, /):
         )
 
 
+@dataclass
+class ReluOpLowering(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, relu: onnx.Relu, rewriter: PatternRewriter, /):
+        body = Region(Block(arg_types=(f64, f64)))
+        zero = arith.Constant(FloatAttr(0, f64))
+        with ImplicitBuilder(body.blocks[0]):
+            max_op = arith.Maximumf(body.blocks[0].args[0], zero.result)
+            linalg.YieldOp(max_op.result)
+        affine_map = AffineMapAttr(AffineMap.from_callable(lambda d0, d1: (d0, d1)))
+        rewriter.replace_matched_op(
+            (
+                empty := tensor.EmptyOp((), relu.res.type),
+                linalg.Generic(
+                    (relu.operand,),
+                    (empty.tensor,),
+                    body,
+                    (
+                        affine_map,
+                        affine_map,
+                    ),
+                    (
+                        linalg.IteratorTypeAttr.parallel(),
+                        linalg.IteratorTypeAttr.parallel(),
+                    ),
+                    (relu.res.type,),
+                ),
+            )
+        )
+
```

Comment:
```suggestion
        body = Region(Block(arg_types=(f64, f64)))
        affine_map = AffineMapAttr(AffineMap.from_callable(lambda d0, d1: (d0, d1)))
        rewriter.replace_matched_op(
            (
                empty := tensor.EmptyOp((), relu.res.type),
                zero := arith.Constant(FloatAttr(0, f64))
                generic := linalg.Generic(
                    (relu.operand,),
                    (empty.tensor,),
                    body,
                    (
                        affine_map,
                        affine_map,
                    ),
                    (
                        linalg.IteratorTypeAttr.parallel(),
                        linalg.IteratorTypeAttr.parallel(),
                    ),
                    (relu.res.type,),
                ),
            )
        )
        with ImplicitBuilder(body) as (in, out):
            max_op = arith.Maximumf(in, zero.result)
            linalg.YieldOp(max_op.result)
```
The zero op was never inserted, also there's a convenience with the ImplicitBuilder of getting the block arguments with the `with` syntax.

Reply:
:)